linera-sdk/src/ethereum.rs

```diff
@@ -5,31 +5,47 @@
 
 use std::fmt::Debug;
 
-use ethers::providers::{JsonRpcClient, ProviderError};
-use serde::{de::DeserializeOwned, Serialize};
+use async_graphql::scalar;
+use async_trait::async_trait;
+use linera_ethereum::{client::JsonRpcClient, common::EthereumServiceError};
+use serde::{Deserialize, Serialize};
 
 use crate::contract::wit::contract_system_api;
 
 /// A wrapper for a URL that implements `JsonRpcClient` and uses the JSON oracle to make requests.
-#[derive(Debug)]
+#[derive(Debug, Default, Deserialize, Serialize)]
 pub struct EthereumClient {
     /// The URL of the JSON-RPC server, without the method or parameters.
     pub url: String,
 }
+scalar!(EthereumClient);
 
-#[cfg_attr(target_arch = "wasm32", async_trait::async_trait(?Send))]
-#[cfg_attr(not(target_arch = "wasm32"), async_trait::async_trait)]
+impl EthereumClient {
+    /// Creates a new `EthereumClient` from an URL.
+    pub fn new(url: String) -> Self {
+        Self { url }
+    }
+}
+
+/// The Ethereum type for a single event
+pub use linera_ethereum::common::EthereumDataType;
+/// The Ethereum type for an event
+pub use linera_ethereum::common::EthereumEvent;
```

Comment:
Let's move all the `pub use` statements to the top, maybe below the private `use` statements?

Reply:
Ok.